python/pyspark/worker.py

```diff
@@ -598,8 +599,18 @@ def process():
             process()
     except Exception:
         try:
+            exc_info = traceback.format_exc()
+            if sys.version_info.major < 3:
+                if isinstance(exc_info, unicode):
+                    exc_info = exc_info.encode("utf-8")
+                else:
```

Comment:
We can actually remove all the version condition and switch this condition to isinstance(... bytes) I guess.

Reply:
I mean:

```python
if isinstance(exc_info, bytes):
    exc_info = exc_info.decode("utf-8", "replace").encode("utf-8")
else:
    exc_info = exc_info.encode("utf-8")
```

`str` in Python 2 is same as `bytes`:

```python
>>> str == bytes
True

>>> isinstance("a", bytes)
True
```

Reply:
This looks better.